packages/web/src/app/admin/products/ProductForm.component.tsx

```diff
@@ -0,0 +1,337 @@
+"use client";
+
+import { FormEvent, useEffect, useState } from "react";
+import { PhotoIcon } from "@heroicons/react/24/solid";
+import RubberDuck from "@/lib/model/rubberduck/Rubberduck.type";
+import Button from "./Button.component";
+import Image from "next/image";
+
+// Define specific types for form dropdowns to ensure type safety.
+type DuckColor = "red" | "yellow" | "green" | "blue";
```

Comment:
Do we want to limit the colors to those 4? I think it will be better if we allow to enter a custom string instead of selecting one out of these four.